osf/management/commands/migrate_preprint_providers.py

```diff
@@ -0,0 +1,86 @@
+import logging
+
+from django.core.management.base import BaseCommand
+
+from osf.models import Preprint, PreprintProvider
+
+
+"""
+A management command to migrate preperints from one provider to another.
+
+i.e. docker-compose run --rm web python3 manage.py migrate_preprint_providers --source_provider lawarxiv --destination_provider osf
+"""
+
+
+logger = logging.getLogger(__name__)
+
+
+def migrate_preprint_providers(source_provider_guid, destination_provider_guid, dry_run=False, delete_source_provider=False):
+    source_provider = PreprintProvider.load(source_provider_guid)
+    destination_provider = PreprintProvider.load(destination_provider_guid)
+    migration_count = 0
+
+    for preprint in Preprint.objects.filter(provider=source_provider):
+        preprint.provider = destination_provider
```

Comment:
This additionally needs to ensure that the subjects from the source provider are mapped to the destination's subjects. Thankfully, we already have helper for that: https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/models/mixins.py#L1130-L1160

However, because it's setting a many-to-many, you'll need to change how `dry_run` is being handled -- 
```
# in `def handle`
with transaction.atomic():
    # Do stuff
    if dry_run:
        raise RuntimeError('Dry run, transaction rolled back')

```